tools/src/main/java/com/hedera/block/tools/commands/ConvertToJson.java

```diff
@@ -0,0 +1,162 @@
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+/**
+ * Command line command that converts binary block stream files to JSON
+ */
+@SuppressWarnings({"DataFlowIssue", "unused", "DuplicatedCode", "ConstantValue"})
+@Command(name = "json", description = "Converts a binary block stream to JSON")
+public class ConvertToJson implements Runnable {
+
+    @Parameters(index = "0..*")
+    private File[] files;
+
+    @SuppressWarnings({"FieldMayBeFinal", "FieldCanBeLocal"})
+    @Option(names = {"-t", "--transactions"}, description = "expand transactions")
+    private boolean expandTransactions = false;
+
+    @SuppressWarnings({"FieldMayBeFinal", "FieldCanBeLocal"})
+    @Option(names = {"-ms", "--min-size"}, description = "minimum file size in megabytes")
+    private double minSizeMb = Double.MAX_VALUE;
+
+    @Override
+    public void run() {
+        if (files == null || files.length == 0) {
+            System.err.println("No files to convert");
+        } else {
+            Arrays.stream(files)
+                    .map(File::toPath)
+                    .flatMap(path -> {
+                        try {
+                            return Files.walk(path);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    .filter(Files::isRegularFile)
+                    .filter(file -> file.getFileName().toString().endsWith(".blk") || file.getFileName().toString()
+                            .endsWith(".blk.gz"))
+                    .filter(file -> { // handle min file size
+                        try {
+                            return minSizeMb == Double.MAX_VALUE || Files.size(file) / 1024.0 / 1024.0 >= minSizeMb;
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    .sorted(Comparator.comparing(file -> file.getFileName().toString()))
+                    .parallel()
+                    .forEach(this::convert);
+        }
+    }
+
+    /**
+     * Converts a binary block stream to JSON
+     *
+     * @param blockProtoFile the binary block stream file
+     */
+    private void convert(Path blockProtoFile) {
+        final String fileName = blockProtoFile.getFileName().toString();
+        final String fileNameNoExt = fileName.substring(0, fileName.lastIndexOf('.'));
+        final Path outputFile = blockProtoFile.resolveSibling(fileNameNoExt + ".json");
+        try (InputStream fIn = Files.newInputStream(blockProtoFile)) {
+            byte[] uncompressedData;
+            if (blockProtoFile.getFileName().toString().endsWith(".gz")) {
+                uncompressedData = new GZIPInputStream(fIn).readAllBytes();
+            } else {
+                uncompressedData = fIn.readAllBytes();
+            }
```

Comment:
Same as in `BlockInfo`:

This branch is currently quite behind, after rebasing you will have access to the `common` module which already has a method that will read file bytes unsafe: `com.hedera.block.common.utils.FileUtilities#readFileBytesUnsafe(Path)` I suggest that you use that.

Reply:
maybe later